server/src/main/java/org/elasticsearch/indices/IndexingMemoryController.java

```diff
@@ -284,8 +324,23 @@ public void bytesWritten(int bytes) {
             }
         }
 
+        public boolean tryRun() {
+            if (runLock.tryLock() == false) {
```

Comment:
mainly style but I find it easier to read if we wrap the try in the if clause and compare `if(runLock.tryLock())` instead